Assets/MixedRealityToolkit/_Core/Devices/BaseSpatialObserver.cs

```diff
@@ -0,0 +1,179 @@
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Devices
+{
+    public class BaseSpatialObserver : IMixedRealitySpatialAwarenessObserver
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="priority"></param>
+        public BaseSpatialObserver(string name, uint priority)
+        {
+            Name = name;
+            Priority = priority;
+        }
+
+        #region IMixedRealityManager implementation
+
+        public string Name { get; }
+
+        /// <inheritdoc />
+        public uint Priority { get; }
+
+        /// <inheritdoc />
+        public virtual void Initialize() { }
+
+        /// <inheritdoc />
+        public virtual void Reset() { }
+
+        /// <inheritdoc />
+        public virtual void Enable() { }
+
+        /// <inheritdoc />
+        public virtual void Update() { }
+
+        /// <inheritdoc />
+        public virtual void Disable() { }
+
+        /// <inheritdoc />
+        public virtual void Destroy() { }
+
+        #endregion IMixedRealityManager implementation
+
+        /// <summary>
+        /// Is the observer running (actively accumulating spatial data)?
+        /// </summary>
+        public bool IsRunning { get; protected set; }
+
+        /// <summary>
+        /// The collection of mesh <see cref="GameObject"/>s that have been observed.
+        /// </summary>
+        public virtual IDictionary<int, GameObject> Meshes => new Dictionary<int, GameObject>();
+
+        /// <summary>
+        /// Start the observer.
+        /// </summary>
+        public virtual void StartObserving()
+        { }
+
+        /// <summary>
+        /// Stop the observer.
+        /// </summary>
+        public virtual void StopObserving()
+        { }
+
+        /// <summary>
+        /// When a mesh is created we will need to create a game object with a minimum 
+        /// set of components to contain the mesh.  These are the required component types.
+        /// </summary>
+        private System.Type[] requiredMeshComponents =
+        {
+            typeof(MeshFilter),
+            typeof(MeshRenderer),
+            typeof(MeshCollider)
+        };
+
+        /// <summary>
+        /// The collection of meshes being managed by the observer.
+        /// </summary>
+        protected Dictionary<int, SpatialMeshObject> meshObjects = new Dictionary<int, SpatialMeshObject>();
+
+        /// <summary>
+        /// Creates a <see cref="SpatialMeshObject"/>.
+        /// </summary>
+        /// <param name="mesh"></param> todo: add comments
+        /// <param name="name"></param>
+        /// <param name="meshId"></param>
+        /// <returns>
+        /// SpatialMeshObject containing the fields that describe the mesh.
+        /// </returns>
+        protected SpatialMeshObject CreateSpatialMeshObject(
```

Comment:
Should this be part of the interface?

Reply:
no. it is an internal observer construct.